JavaScript/Topics/3.Web_Browser/README.md

```diff
@@ -1 +1,176 @@
+
+![error_internal_js](Code_Snippets/1.Error_Internal_Js.png)
+
+After adding the `window.onload` function:
+
+``` HTML
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Internal JS</title>
+
+    <script>
+        "use strict";
+        window.onload = function what() {
+            let x = document.getElementById("main");
+            x.innerHTML = "Internal JavaScript";
+        };
+    </script>
+</head>
+<body>
+    <h1 id = "main"></h1>
+</body>
+</html>
+
+```
+
+The output is shown below, after running the HTML file.
+![Output](Code_Snippets/2.Output_Internal_Js.png)
+
+## 2. External JavaScript
+The most reliable approach to construct the webpage is to write the JavaScript code in a separate file. It should always be a practice to write JavaScript code in the separate file, so that when a programmer creates a large block of code, they can quickly traverse the code snippets in the separate file rather than scrolling through the HTML page.
+
+The following code snippet is added inside the header content to link JavaScript with the HTML file:
+
+`<script src = "file_name.js"></script>`
+
+The js file is shown below:<br>
+``` JS
+
+"use strict";
+window.onload = function what() {
+    let x = document.getElementById("main");
+    x.innerHTML = "External JavaScript";
+};
+
+```
+
+The output is shown below, after running the HTML file.<br>
+![Output External JS](Code_Snippets/3.Output_External_Js.png)
+
+<br>
+
+<b>Note: </b><br>
+If there is no *window.onload* function in JS file, add the `<script>` tag at the end of the body content. Otherwise it will load before the body content and throw an error.
+
+```HTML
+
+<body>
+    <h1 id = "main"></h1>
+
+    <script src = "../JS/script.js"></script>
+</body>
+
+```
+Else, it can be added in the head content and JS can access the DOM elements if `window.onload` function is included.
+
+<br><hr><br>
+
+# How JavaScript files get compiled?
+
+In most low-level languages, the files get compiled and then executed accordingly. But in JavaScript, compilation takes place during the execution of the file. This type of compiler is known as Just-in-time(JIT) compiler.
+
+For example, V8 engine is used for compilation of JavaScript files. It is developed by Google.
+
+<br>
+
+## Working of JIT compiler
+
+The Source code of JS file is converted into the *byte code*, which is the intermediate conversion.<br> 
+This bytecode gets interpreted by Virtual Machine. Then it is converted to machine understandable code in section-wise (can be per function or per line) dynamically so that the program can run faster.
+
```

Comment:
Add image to show JIT.
https://duckduckgo.com/?t=lm&q=JIT+Compilation+javascript&atb=v289-4gy&iax=images&ia=images&iai=https%3A%2F%2Fmiro.medium.com%2Fmax%2F1200%2F1*SezjSMIo5Zub4SyCfZ9iKA.png

If you find some other nice picture then neglect mine and add yours

Reply:
Should I include the source of this image?

Reply:
yes